crates/symbolicator-service/src/download/mod.rs

```diff
@@ -101,15 +101,18 @@ type CountedFailures = Arc<Mutex<VecDeque<FailureCount>>>;
 /// A structure that keeps track of download failures in a given time interval
 /// and puts hosts on a block list accordingly.
 ///
-/// The logic works like this: if a host has at least `FAILURE_THRESHOLD` download
-/// failures in a window of `TIME_WINDOW` seconds, it will be blocked for a duration of
-/// `BLOCK_TIME`.
+/// The logic works like this: if a host has at least `failure_threshold` download
+/// failures in a window of `time_window_millis` ms, it will be blocked for a duration of
+/// `block_time`.
+///
+/// Hosts included in `never_block` will never be blocked regardless of download_failures.
 #[derive(Clone, Debug)]
 struct HostDenyList {
     time_window_millis: u64,
     bucket_size_millis: u64,
     failure_threshold: usize,
     block_time: Duration,
+    never_block: Vec<String>,
```

Comment:
Maybe we should make this optional as well?

Reply:
How do you mean? Make it `Option<Vec<String>>`? I don't think there's a meaningful difference between `None` and an empty `Vec`.

Reply:
I assume empty vector makes more allocations then None. Is it not?

Reply:
It would come down to the difference between
```rust
if !self.never_block.contains(&host) {
    […]
}
```
and something like
```rust
if self.never_block.map_or(true, |nb| !nb.contains(&host)) {
    […]
}
```
Most of the time I would expect `never_block` to be empty or very short, so I'm not sure it's worth adding a short-circuit via `None`.